src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java

```diff
@@ -85,6 +86,15 @@
         parameterMap.put(
                 ArPropertyConstants.ACCOUNT_DETAILS + "." + KFSPropertyConstants.CONTRACT_CONTROL_ACCOUNT_NUMBER,
                 getRecipientAccountNumber(document.getAccountDetails()));
+        
+        if (document.getInvoiceGeneralDetail().isFinalBillIndicator()) {
+            parameterMap.put(CuArPropertyConstants.ContractsAndGrantsBillingAwardFields.FINAL_BILL, "X");
+            parameterMap.put(CuArPropertyConstants.ContractsAndGrantsBillingAwardFields.PARTIAL_BILL, "");
```

Comment:
I would suggest using constants for the empty string and the "X" value.

Reply:
OK, can do